### PR TITLE
fixed issue #3299 (hide mark all as read button for no unread notifications

### LINKED ISSUE #3299: fix: not to show mark all as read button for no unread notifications.
**Describe the bug**
not to show mark all as read button for no unread notifications.

**To Reproduce**
1. Go to 'https://circuitverse.org/'
2. Click on 'notification icon'
3. Scroll down to 'see all'

**Expected behavior**
The button for `mark all as read` should be visible when there are unread notifications.

**Screenshots**
Screenshot from 2022-09-28 08-36-14

**Device Information [optional]:**

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by r3yc0n1c**
Fixes #3299

#### Describe the changes you have made in this PR -
- hide the `Mark all as read` button if there is no unread notification

### Screenshots of the changes (If any) -
| Before  | After  |
|:---:|:---:|
| b4  | after|

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -20,7 +19,7 @@
     t.bigint "user_id"
     t.string "name"
     t.json "properties"
-    t.datetime "time"
+    t.datetime "time", precision: nil
```
Comment (vedant-jain03) [inline db/schema.rb:22]: These are irrelevant changes!

File: app/views/users/noticed_notifications/index.html.erb [hunk 2]
```diff
@@ -11,9 +11,13 @@
...
-    <div>
-      <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text" %>
-    </div>
+
+    <% if @unread.count != 0 %>
+      <div>
+        <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text" %>
+      </div>
+    <% end %>
+
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/views/users/noticed_notifications/index.html.erb:20]: @r3yc0n1c Resolve this issue!

File: app/views/users/noticed_notifications/index.html.erb [hunk 3]
```diff
@@ -11,9 +11,13 @@
       <span id="all-notifications" class="active notifications-tab" data-action="click->notifications#activeAllNotifications"><%= t("users.notifications.all") %></span>
       <span id="unread-notifications" class="notifications-tab" data-action="click->notifications#activeUnreadNotifications"><%= t("users.notifications.unread") %></span>
     </div>
-    <div>
-      <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text" %>
-    </div>
+
```
Comment (vedant-jain03) [inline app/views/users/noticed_notifications/index.html.erb:14]: Remove this extra link as well!

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

File: app/views/users/noticed_notifications/index.html.erb [hunk 4]
```diff
@@ -11,9 +11,13 @@
...
+    <% if @unread.count != 0 %>
```
Comment (tachyons) [inline app/views/users/noticed_notifications/index.html.erb:15]:
```suggestion
    <% if @unread.positive? %>
```

Comment (r3yc0n1c) [reply app/views/users/noticed_notifications/index.html.erb:15]: it looks better now... thank you! shall I commit the changes?

Comment (vedant-jain03) [reply app/views/users/noticed_notifications/index.html.erb:15]: yes, could you also please remove the extra lines mentioned by hound!

File: app/views/users/noticed_notifications/index.html.erb [hunk 5]
```diff
@@ -11,9 +11,11 @@
       <span id="all-notifications" class="active notifications-tab" data-action="click->notifications#activeAllNotifications"><%= t("users.notifications.all") %></span>
       <span id="unread-notifications" class="notifications-tab" data-action="click->notifications#activeUnreadNotifications"><%= t("users.notifications.unread") %></span>
     </div>
-    <div>
-      <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text" %>
-    </div>
+    <% if @unread.positive? %>
```
Comment (tachyons) [inline app/views/users/noticed_notifications/index.html.erb:14]:
```suggestion
    <% if @unread.count.positive? %>
```

**Pull request merged**